### DISCUSSION TITLE
Getting the UFS Weather Model to output physics tendencies

### DISCUSSION THREAD

**Discussion by gsketefian** [ufs-community/ufs-srweather-app#862] [Q&A]
Hi,

I am trying to get the UFS Weather Model (running through the UFS SRW App) to output physics tendencies.  From [this page](https://ccpp-techdoc.readthedocs.io/en/latest/ParamSpecificOutput.html), it seems the way to do this is to change the `ldiag3d` and `qdiag3d` namelist options to `.true.`.  This should cause the model to output the `fv3_history2d.nc` and/or `fv3_history.nc` files that will contain the physics tendencies. So I changed `ldiag3d` and `qdiag3d` to `.true.` in `parm/input.nml.FV3` and reran my cases (which are just 4 of the workflow end-to-end tests), but there was no change in the model output, i.e. the `fv3_history2d.nc` and `fv3_history.nc` files were not generated.  I checked the `input.nml` files for my experiments, and those do indeed contain `ldiag3d` and `qdiag3d` set to `.true.` (so that part is working right).  Is there something else (maybe in `diag_table`) that I need to change to get the model to output those history files?  Thanks.

**Comment by gspetro-NOAA** [comment]
Hi Gerard,

Not sure if you've seen WM Discussions [#1862](https://github.com/ufs-community/ufs-weather-model/discussions/1862) and [#1867](https://github.com/ufs-community/ufs-weather-model/discussions/1867), but it looks like there are changes that need to be made to the `diag_table`. The WM has documentation [here](https://ccpp-techdoc.readthedocs.io/en/latest/ParamSpecificOutput.html#outputting-tendencies) and [here](https://ccpp-techdoc.readthedocs.io/en/latest/ParamSpecificOutput.html#output-of-auxiliary-arrays-from-ccpp) on the topic. If you manage(d) to get this working, I think @aschuh would be interested to hear how. Otherwise, you can follow his two discussions, since I think it makes the most sense to handle this question in the WM repo. 

Best,
Gillian

**Reply by gspetro-NOAA** [reply]
It looks like the physics tendencies will be output in `sfc*.nc` files once the `diag_table` changes have been made (see links above for information on how to do this). Users have been a bit confused by the output location, so I thought I'd mention it here. :)

**Comment by gsketefian** [comment]
@gspetro-NOAA Oh ok.  Even if they're 3-D fields and/or not related to the surface?  @swomantree and I ended up turning off quilting and looking at the tendencies on the native grid (i.e. in the `fv3_history.nc` file).  I'm actually not familiar with the `sfc*.nc` files (I should read the other discussions but haven't gotten around to it yet).

**Reply by gspetro-NOAA** [reply]
@gsketefian Yeah, everyone I've talked to agrees that it doesn't make sense, but that is where the 3-D fields seem to show up! I believe they'll be in there even if not related to the surface, but if any variables seem to be missing, make sure to check other 2-D files (even though they're 3-D fields) before throwing up your hands in confusion. ;)

**Comment by swomantree** [comment]
@gsketefian <https://github.com/gsketefian> I wonder if your bug fix for
outputting the physics tendencies would be helpful for this discussion
<https://github.com/ufs-community/ufs-weather-model/discussions/1867>. I
know you helped me fix 4 lines of code so I was able to output those
fv3_history files, and I thought you had made a separate bug/issue report
detailing what you did. I think it's
<https://github.com/ufs-community/ufs-srweather-app/pull/869>this
<https://github.com/ufs-community/ufs-srweather-app/pull/869>fix that did
the trick.
 The last post on that discussion said that he tried to turn off quilting
to output those tendencies, but the model crashed.  That happened to us as
well, before we turned the ldiag3d and qdiag3d namelist options in the
weather model's namelist on and set quilting to false.
